### DISCUSSION TITLE
Intel's Vulkan Linux Driver Adds H.265 Video Decoding Support

### DISCUSSION THREAD

**Discussion by alatteri** [CESNET/UltraGrid#314] [Ideas]
https://www.phoronix.com/news/Intel-ANV-Vulkan-Video-H265

This might be useful.  Don't know if the capability comes automatically with the mesa update.

**Comment by alatteri** [comment]
FFmpeg's Next Release Will Be Exciting With Vulkan Video Decode, More Vulkan Filters

https://www.phoronix.com/news/FFmpeg-Vulkan-Video

**Comment by alatteri** [comment]
RADV Vulkan Video Gets More Robust H.265 Decoding

https://www.phoronix.com/news/RADV-More-Robust-H265-HEVC

**Comment by MartinPulec** [comment]
Hi @alatteri, thanks for all the info. I believe that the support for Vuklan decode will arrive at some time point in future.

**Comment by alatteri** [comment]
Will we automatically get this with the updates in FFmpeg?

### From FFmpeg.org
> May 31st, 2023, Vulkan decoding
> A few days ago, Vulkan-powered decoding hardware acceleration code was merged into the codebase. This is the first vendor-generic and platform-generic decode acceleration API, enabling the same code to be used on multiple platforms, with very minimal overhead. This is also the first multi-threaded hardware decoding API, and our code makes full use of this, saturating all available decode engines the hardware exposes.
> 
> Those wishing to test the code can read our [documentation page](https://trac.ffmpeg.org/wiki/HWAccelIntro#Vulkan). For those who would like to integrate FFmpeg's Vulkan code to demux, parse, decode, and receive a VkImage to present or manipulate, documentation and examples are available in our source tree. Currently, using the latest available git checkout of our [repository](https://git.videolan.org/?p=ffmpeg.git;a=summary) is required. The functionality will be included in stable branches with the release of version 6.1, due to be released soon.
> 
> As this is also the first practical implementation of the specifications, bugs may be present, particularly in drivers, and, although passing verification, the implementation itself. New codecs, and encoding support are also being worked on, by both the Khronos organization for standardizing, and us as implementing it, and giving feedback on improving.

**Comment by mpiatka** [comment]
Well, not automatically, but this is something we're looking to implement.
Note however, that H.265 decoding requires the Mesa library version 23.2 which is currently not even released yet and then will take some more time until it's included in linux distributions. So it'll probably take some time before it gets usable.

**Comment by mpiatka** [comment]
I've just added a basic support for ffmpeg's Vulkan decode. I've tested it on archlinux with i7-11800 (Intel TigerLake-H GT1 GPU)  and it appeared to work. However, to get it to run I had to set the `ANV_VIDEO_DECODE=1` flag and to get HEVC working I had to compile mesa from source.

I also tried it on Nvidia RTX 3060 (mobile) and that appeared to be working as well, but in practice it won't be selected as the vdpau and cuda accelerations have higher priority.
